driver-core/src/main/com/mongodb/client/model/expressions/MqlExpression.java

```diff
@@ -137,6 +137,99 @@ public <R extends Expression> R cond(final R left, final R right) {
         return newMqlExpression(ast("$cond", left, right));
     }
 
+    /** @see DocumentExpression */
+
+    private Function<CodecRegistry, AstPlaceholder> getFieldInternal(final String fieldBoolean) {
+        // TODO these "getFieldX" names are all very long, and longer with composite paths
+        return (cr) -> astDoc("$getField", new BsonDocument()
+                .append("input", this.fn.apply(cr).bsonValue)
+                .append("field", new BsonString(fieldBoolean)));
+    }
+
+    @Override
+    public BooleanExpression getBoolean(final String field) {
```

Comment:
I have renamed `getFieldBoolean` to `getBoolean` (same for other types). This seems nicer, and incidentally similar to jdbc resultset.

We could also rename setField->set and unsetField->unset, but need to see how this turns out wrt other method names.